src/pudl_archiver/archivers/eia/eiarecs.py

```diff
@@ -0,0 +1,115 @@
+"""Archive EIA Residential Energy Consumption Survey (RECS)."""
+
+import logging
+import re
+
+from pudl_archiver.archivers.classes import (
+    AbstractDatasetArchiver,
+    ArchiveAwaitable,
+    ResourceInfo,
+)
+from pudl_archiver.frictionless import ZipLayout
+
+LINK_PATTERNS = [
+    # housing characteristics
+    {
+        "base_url": "https://www.eia.gov/consumption/residential/data",
+        "php_extension": "index.php?view=characteristics",
```

Comment:
The RECS [webpage](https://www.eia.gov/consumption/residential/data/2020/) is a little bit tricky. It has tabs containing different sequences of datafiles (and PDFs, etc.) that can't be reached directly from the base_url. Instead we need to tack on these strings to the end of the url, which are different for different tabs.

Reply:
Might be worth defining the shape of this dictionary as a dataclass, but definitely not a blocking concern.